k3k-kubelet/main.go

```diff
@@ -128,7 +139,7 @@ func run(clx *cli.Context) error {
 		logger.Fatalw("failed to create new virtual kubelet instance", zap.Error(err))
 	}
 
-	if err := k.registerNode(ctx, k.agentIP, cfg.KubeletPort, cfg.ClusterNamespace, cfg.ClusterName, cfg.AgentHostname, cfg.ServerIP, k.dnsIP, cfg.Version); err != nil {
+	if err := k.registerNode(ctx, k.agentIP, cfg.KubeletPort, cfg.ClusterNamespace, cfg.ClusterName, cfg.AgentHostname, cfg.ServerIP, k.dnsIP, cfg.Version, cfg.MirrorHostNodes); err != nil {
```

Comment:
Maybe we can just pass the `cfg` instead of adding one more argument (10 are a lot!).

Reply:
yes this was an old code and we kept adding fields to it, I will fix